src/openrct2/scripting/bindings/ride/ScRide.cpp

```diff
@@ -524,6 +524,72 @@ namespace OpenRCT2::Scripting
         return ride != nullptr ? ride->GetRideTypeDescriptor().LiftData.minimum_speed : 0;
     }
 
+    int32_t ScRide::maxSpeed_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ((ride->max_speed) * 9) >> 18 : 0;
+    }
+
+    int32_t ScRide::averageSpeed_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ((ride->average_speed) * 9) >> 18 : 0;
+    }
+
+    int32_t ScRide::rideTime_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->GetTotalTime() : 0;
+    }
+
+    int32_t ScRide::rideLength_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->GetTotalLength() >> 16 : 0;
+    }
+
+    int32_t ScRide::maxPositiveVerticalGs_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->max_positive_vertical_g : 0;
+    }
+
+    int32_t ScRide::maxNegativeVerticalGs_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->max_negative_vertical_g : 0;
+    }
+
+    int32_t ScRide::maxLateralGs_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->max_lateral_g : 0;
+    }
+
+    uint16_t ScRide::totalAirTime_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->total_air_time * 3 : 0;
+    }
+
+    uint8_t ScRide::drops_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->drops & 0b00111111 : 0;
+    }
+
+    uint8_t ScRide::poweredLifts_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->drops & 0b11000000 : 0;
+    }
+
+    uint8_t ScRide::highestDropHeight_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? (ride->highest_drop_height >> 1) + (ride->highest_drop_height >> 2) : 0;
```

Comment:
Aren't we loosing some accuracy here as well?

Reply:
Same as above.